Fixes for blob store

jdrossl opened the pull request:
https://github.com/craftercms/craftercms/issues/4669

jdrossl committed:
Revert "Fixed blob issue"

This reverts commit 1fac146dabef337e8b1f13a6c9397b1388608d18.

jdrossl committed:
Fixes for blob store

sumerjabri reviewed (changes requested):
Please switch the data structures to treemap and update GitContentRepo in the two spots we discussed.

jdrossl committed:
Optimizations for git processing:
- Use TreeMap instead of HashMap
- Remove unused variables
- Remove unnecessary jgit operations
